MoreLinq/Batch.cs

```diff
@@ -87,69 +86,65 @@ public static IEnumerable<IEnumerable<TSource>> Batch<TSource>(this IEnumerable<
             if (size <= 0) throw new ArgumentOutOfRangeException(nameof(size));
             if (resultSelector == null) throw new ArgumentNullException(nameof(resultSelector));
 
-            switch (source)
+            return Impl(); IEnumerable<TResult> Impl()
```

Comment:
Used per updated convention set in #944 (and other operators prior to #944). Either a) `_()` is desired, and `ExceptBy`, `Interleave`, `InsertBy`, `EndsWith`, `Await`, and `SortedMerge` should be updated to reflect (and any difficulties with discards should be addressed), or b) a PR should be done to update all operators to `Impl`. 

Either way, currently, both `_` and `Impl` are present enough to be considered convention. Would recommend that this remains as `Impl`, and an issue opened to return at a later date to update all operators to use `Impl`.

Reply:
Yes, the code base is inconsistent at the moment. `Impl` was used where absolutely necessary due to C# discards quirk-mode. The _majority_ convention was `_` before #944 when an iterator could be a mere local function. In general, the `_` convention for an iterator will need a bigger reconsideration with #906 when even `Impl` won't be a good choice since there won't be much motivation left to keep the iterator as a local function (it could be just lifted out as a stand-alone static method again and then potentially need a unique name).

There is no need to go with `Impl` here so the idea was to keep the majority convention for now.